compatibility layer fails to get item-type for new QuantityType items

item-configuration:

```
Number:Temperature AMSTestAussenTemperatur "Aussen [%.2f °C]" (gAussen, gPersist5) { mqtt = ">[mqtt:sensor/temperatur/aussentest:state:*:default]" }
```

## Expected Behaviour

value gets send to MQTT topic

## Current Behavior
```
12:59:12.217 [ERROR] [del.item.internal.GenericItemProvider] - Binding configuration of type 'mqtt' of item 'AMSTestAussenTemperatur' could not be parsed correctly.
org.eclipse.smarthome.model.item.BindingConfigParseException: cannot process unknown item type Number:Temperature
        at org.openhab.core.binding.internal.BindingConfigReaderDelegate.getOpenHABItem(BindingConfigReaderDelegate.java:87) ~[?:?]
        at org.openhab.core.binding.internal.BindingConfigReaderDelegate.getOpenHABItem(BindingConfigReaderDelegate.java:57) ~[?:?]
        at org.openhab.core.binding.internal.BindingConfigReaderDelegate.validateItemType(BindingConfigReaderDelegate.java:38) ~[?:?]
        at org.eclipse.smarthome.model.item.internal.GenericItemProvider.internalDispatchBindings(GenericItemProvider.java:360) ~[?:?]
        at org.eclipse.smarthome.model.item.internal.GenericItemProvider.internalDispatchBindings(GenericItemProvider.java:330) ~[?:?]
        at org.eclipse.smarthome.model.item.internal.GenericItemProvider.processBindingConfigsFromModel(GenericItemProvider.java:204) ~[?:?]
        at org.eclipse.smarthome.model.item.internal.GenericItemProvider.modelChanged(GenericItemProvider.java:397) ~[?:?]
        at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.notifyListeners(ModelRepositoryImpl.java:301) ~[?:?]
        at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.addOrRefreshModel(ModelRepositoryImpl.java:139) ~[?:?]
        at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.checkFile(FolderObserver.java:227) ~[?:?]
        at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.processWatchEvent(FolderObserver.java:291) ~[?:?]
        at org.eclipse.smarthome.core.service.WatchQueueReader.lambda$3(WatchQueueReader.java:323) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
```

## Possible solution

Use only first part of item-type (before ":") in [https://github.com/openhab/openhab-core/blob/master/bundles/org.openhab.core.compat1x/src/main/java/org/openhab/core/binding/internal/BindingConfigReaderDelegate.java#L63](https://github.com/openhab/openhab-core/blob/master/bundles/org.openhab.core.compat1x/src/main/java/org/openhab/core/binding/internal/BindingConfigReaderDelegate.java#L63)

--- a/bundles/org.openhab.core.compat1x/src/main/java/org/openhab/core/binding/internal/BindingConfigReaderDelegate.java
+++ b/bundles/org.openhab.core.compat1x/src/main/java/org/openhab/core/binding/internal/BindingConfigReaderDelegate.java
@@ -9,6 +9,7 @@
 package org.openhab.core.binding.internal;
 
 import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.items.ItemUtil;
 import org.eclipse.smarthome.model.item.BindingConfigParseException;
 import org.eclipse.smarthome.model.item.BindingConfigReader;
 
@@ -60,7 +61,9 @@ public class BindingConfigReaderDelegate implements BindingConfigReader {
     private org.openhab.core.items.Item getOpenHABItem(String itemType, String itemName)
             throws BindingConfigParseException {
 
-        switch (itemType) {
+        String mainType = ItemUtil.getMainItemType(itemType);
+
+        switch (mainType) {
             case "Group":
                 return new org.openhab.core.items.GroupItem(itemName);
             case "Switch":

